golangci-lint: fix issues for contextcheck and turn on

Fix:

```
        internal/librarian/generate.go:100:53: Function `detectIfLibraryConfigured` should pass the context parameter (contextcheck)
        	libraryConfigured, err := detectIfLibraryConfigured(cfg.APIPath, cfg.RepoURL, cfg.RepoRoot, cfg.GitHubToken)
        	                                                   ^
```
Add to enable list in .golangci.yaml.

--- a/.golangci.yaml
+++ b/.golangci.yaml
@@ -26,3 +26,4 @@ linters:
     - staticcheck
     - unparam
     - unused
+    - contextcheck

--- a/internal/librarian/generate.go
+++ b/internal/librarian/generate.go
@@ -97,7 +97,7 @@ func init() {
 }
 
 func runGenerate(ctx context.Context, cfg *config.Config) error {
-	libraryConfigured, err := detectIfLibraryConfigured(cfg.APIPath, cfg.RepoURL, cfg.RepoRoot, cfg.GitHubToken)
+	libraryConfigured, err := detectIfLibraryConfigured(ctx, cfg.APIPath, cfg.RepoURL, cfg.RepoRoot, cfg.GitHubToken)
 	if err != nil {
 		return err
 	}
@@ -207,7 +207,7 @@ func runGenerateCommand(state *commandState, apiRoot, apiPath, outputDir string)
 // pipeline state if repoRoot has been specified, or the remote pipeline state (just
 // by fetching the single file) if flatRepoUrl has been specified. If neither the repo
 // root not the repo url has been specified, we always perform raw generation.
-func detectIfLibraryConfigured(apiPath, repoURL, repoRoot, gitHubToken string) (bool, error) {
+func detectIfLibraryConfigured(ctx context.Context, apiPath, repoURL, repoRoot, gitHubToken string) (bool, error) {
 	if repoURL == "" && repoRoot == "" {
 		slog.Warn("repo url and root are not specified, cannot check if library exists")
 		return false, nil
@@ -232,7 +232,7 @@ func detectIfLibraryConfigured(apiPath, repoURL, repoRoot, gitHubToken string) (
 			slog.Warn("failed to parse", "repo url:", repoURL, "error", err)
 			return false, err
 		}
-		pipelineState, err = fetchRemotePipelineState(context.Background(), languageRepoMetadata, "HEAD", gitHubToken)
+		pipelineState, err = fetchRemotePipelineState(ctx, languageRepoMetadata, "HEAD", gitHubToken)
 		if err != nil {
 			return false, err
 		}

